morello/codegen/ctensors.py

```diff
@@ -23,11 +25,28 @@
     (Uint8, 16): (16, "vub16", "__m128i", ("_mm_loadu_si128", "_mm_storeu_si128")),
 }
 
+_ARM_VEC_TYPES: dict[tuple[Dtype, int], tuple[int, str, str, tuple[str, str]]] = {
+    (Uint32, 8): (32, "vui8", "uint32x4x2_t", ("vld1q_u32", "vst1q_u32")),
+    (Uint32, 4): (16, "vui4", "uint32x4_t", ("vld1_u32", "vst1_u32")),
+    (Uint8, 32): (32, "vub32", "uint32x4x2_t", ("vld1q_u32", "vst1q_u32")),
+    (Uint8, 16): (16, "vub16", "uint32x4_t", ("vld1_u32", "vst1_u32")),
+}
```

Comment:
I am unsure if the instruction conversions from `immintrin.h` to `arm_neon.h` is correct, but apart from that, this PR seems to be ready.

Reply:
Thank you for reminding me about the tests. I will write test codes.

Reply:
That'd be ideal, but it would be fine to just test these manually for this PR.

Reply:
@samkaufman I apologize for the late reply. I conducted a test and can confirm that the current intrinsics are functioning properly on this repository: https://github.com/ken-matsui/arm-neon-example.